common/src/main/kotlin/entity/Interactions.kt

```diff
@@ -391,6 +410,51 @@ sealed class InteractionResponseType(val type: Int) {
         override fun serialize(encoder: Encoder, value: InteractionResponseType) {
             encoder.encodeInt(value.type)
         }
+    }
+}
+
+@KordPreview
+@Serializable
+data class DiscordApplicationCommandPermissions(
+    val id: String,
+    @SerialName("application_id")
+    val applicationId: String,
+    @SerialName("guild_id")
+    val guildId: String,
+    val permissions: List<DiscordApplicationCommandPermission>
+)
 
+@KordPreview
+@Serializable
+data class PartialDiscordApplicationCommandPermissions(
+    val id: String,
+    val permissions: List<DiscordApplicationCommandPermission>
+)
+
+@KordPreview
+@Serializable
+data class DiscordApplicationCommandPermission(
+    val id: Snowflake,
+    val type: Type,
+    val permission: Boolean
+) {
+    @Serializable(with = Type.Serializer::class)
+    enum class Type(val value: Int) {
+        ROLE(1),
+        USER(2);
+
```

Comment:
I'd say a class with `object`s would be more useful than an enum in case Discord, somewhat introduced a new type of permissions